### DISCUSSION TITLE
`use(serverFunction())` - Experimental "use" hook and server action / server function for data fetching

### DISCUSSION THREAD

**Discussion by lsagetlethias** [vercel/next.js#51041] [App Router]
Hello 👋 

My question is pretty simple, I want to use server function for data fetching instead of fetch, to avoid dealing with api endpoint myself.
For that, `use()` experimental hook (with `Suspense` wrapper or `loading.tsx` page) does the job with a fetch, but not with a server function (which should be a promise, but meh, something under the hood of next tell me it's not...).
Yes, I could use `useEffect/useState` instead of `use` but I don't want to deal with undefined values or loading state in this part of the code as well.

Here's a pretty simple exemple:
```tsx
// page.tsx
import { Suspense } from 'react';
import { ClientComponent } from './ClientComponent';

export default function ActionBugPage() {
    return <>
        <h1>Action Bug Page</h1>
        <Suspense><ClientComponent /></Suspense>
    </>;
}

// ClientComponent.tsx
"use client";

import { use, useEffect, useState } from 'react';
import { getData } from './actions';

async function fetchData() {
    console.log("FETCH DATA");
    return await getData();
}
export const ClientComponent = () => {
    const [mounted, setMounted] = useState(false);

    useEffect(() => {
        setMounted(true)
    }, []);


    // emulate sessionStorage loading.
    // getData() server function need a parameters only available after loading
    if (!mounted) return <>NOT MOUNTED</>;
    
    const data = use(fetchData());

    console.log({data});

    return <>{data}</>
}

// actions.ts
"use server";

export async function getData() {
    console.log("Server function called")
    return Promise.resolve("hello");
}
```

<img width="1204" alt="image" src="https://github.com/vercel/next.js/assets/5783789/376d0440-4a39-426a-86a3-9313c3db3f71">

By the way, `use(serverFunction())` isn't working either without the "mounted" condition and that's ok because it would create a fetch waterfall.

**Comment by iojcde** [comment]
Having similar issues here...
Has anyone found a solution to this?
